### PR TITLE
[risk=low][RW-7973] UI for Async Create Workspace

### PR DESCRIPTION
**Pull request by jmthibault79**
Builds on #6375 

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 1]
```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
       this.saveWorkspace();
     }
 
+    private async apiCreateWorkspaceAsync() {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1057]: this is the MVP UI, which the user should experience as having the same behavior as the synchronous version

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 2]
```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
...
+      let pollTimedOut = false;
+      setTimeout(() => (pollTimedOut = true), CREATE_WORKSPACE_TIMEOUT_MS);
+      let workspaceOp = await workspacesApi().createWorkspaceAsync(
+        this.state.workspace
+      );
+      while (
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.tsx:1063]: createWorkspaceAsync() also returns an operation, so we don't need that first call

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 3]
```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
...
+        workspaceOp.status === WorkspaceOperationStatus.PENDING
+      ) {
+        await new Promise((resolve) =>
+          setTimeout(resolve, CREATE_WORKSPACE_INTERVAL_MS)
+        );
+        workspaceOp = await workspacesApi().getWorkspaceOperation(
+          workspaceOp.id
+        );
+      }
+      if (workspaceOp.status === WorkspaceOperationStatus.ERROR) {
```
Comment (dmohs) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1074]:
```suggestion
      if (workspaceOp.status !== WorkspaceOperationStatus.SUCCESS) {
```
This way you would throw the error also in the case where the operation timed out.

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 4]
```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
...
+      let workspaceOp = await workspacesApi().createWorkspaceAsync(
+        this.state.workspace
+      );
+      while (
+        !pollTimedOut &&
+        workspaceOp.status === WorkspaceOperationStatus.PENDING
+      ) {
+        await new Promise((resolve) =>
+          setTimeout(resolve, CREATE_WORKSPACE_INTERVAL_MS)
+        );
```
Comment (dmohs) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1069]: If you import `delay` from subscribable,
```suggestion
        await delay(CREATE_WORKSPACE_INTERVAL_MS);
```

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 5]
```diff
@@ -231,6 +233,10 @@ const DEFAULT_ACCESS_TIER = AccessTierShortNames.Registered;
 const NEW_ACL_DELAY_POLL_TIMEOUT_MS = 60 * 1000;
 const NEW_ACL_DELAY_POLL_INTERVAL_MS = 10 * 1000;
 
+// Poll parameters for checking the result of an async Workspace Create operation
+const CREATE_WORKSPACE_TIMEOUT_MS = 3 * 60 * 1000;
+const CREATE_WORKSPACE_INTERVAL_MS = 5 * 1000;
```
Comment (dmohs) [inline ui/src/app/pages/workspace/workspace-edit.tsx:238]: Maybe?
```suggestion
const CREATE_WORKSPACE_POLL_INTERVAL_MS = 5 * 1000;
```

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.tsx:1069]: nice, wasn't aware of this

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.tsx:1074]: good catch, thanks

File: ui/src/app/utils/subscribable.spec.tsx [hunk 6]
```diff
@@ -1,8 +1,4 @@
-import { atom } from './subscribable';
-
-export const delay = (ms) => {
```
Comment (jmthibault79) [inline ui/src/app/utils/subscribable.spec.tsx:3]: unrelated - heh, this test was not actually testing the `delay` from `subscribable`

**Pull request merged**